frontend/shared-components/state-input/state-input.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FormEvent, useEffect, useState } from "react"
+import styles from "./state-input.module.css"
+import { FormLevelError } from '../index'
+import { capitalizeFirstChar } from '../../helpers/syntax-helper'
+import { states } from '../../models'
+
+interface USAStateInputProps { isSubmitted: boolean }
+export default function USAStateInput({ isSubmitted }: USAStateInputProps) {
+  const {stateSelect, hasError} = styles
+  const [inputId, listId, errorId] = ['stateSelectInput', 'stateSelectList', 'stateSelectError']
```

Comment:
Why are these variables being declared thusly? This is very confusing.

Reply:
Ah okay - I figured 3-1 line brevity would be worthwhile here and I didn't think it'd be that confusing (presumably any React devs would be pretty familiar with array deconstruction)

But I'm not married to the convention - happy to swap it out for regular 1:1 declarations! 

OR looking again at it there is a fair bit of repetition between the names, we could also DRY it up by writing it out `[inputId, listId, errorId] = ['Input', 'List', 'Error'].map(word => 'stateSelect' + word)

ALSO noticing that it probably shouldn't be "stateSelect" anymore as it's no longer a select element - I'll update that as well!

Reply:
I wouldn't go as far as to say 'declare them the typical way': I think a short comment near the top of any component that's declaring vars using this convention would fully address my readability/comprehension concerns. The code works and is more succinct than the alternative, but it is somewhat unorthodox.

Reply:
This array deconstruction is a fairly common construct-- any time you're using `const [x, setX] = useState(z)` you're using that construct!

Reply:
Hmm so I'm pretty against the idea of using comments for something so inconsequential - I'm also not as keen on my other idea of mapping (think it complicates things unnecessarily) 

So I think the viable options are either keeping it as is or doing the 1:1 way - @dylanesque any preference?